Android/app/src/go/intra/split/retrier.go

```diff
@@ -234,7 +247,55 @@ func splitHello(hello []byte) ([]byte, []byte) {
 	if s > limit {
 		s = limit
 	}
-	return hello[:s], hello[s:]
+	splitLen = int16(s)
+	pkts = net.Buffers{hello[:s], hello[s:]}
```

Comment:
You can do the split without extra allocations:
- Save `recordLen`
- Set `hello[3:5]` to the split record len
- Write `hello[:split]` (new first record)
- Copy record header `hello[0:5]` to `hello[split-5:split]`
- Update record len in the new header
- Write `hello[split-5:]` (new second record)

Reply:
If you would like to decouple the TCP split from the record fragmentation, you can make the writer be `split.NewWriter(w, tcpSplit)`. Though I suggest we align the TCP segment with the TLS record for now to minimize things changing.

Reply:
I will just fragment at the split point and not doing any TCP split any more (because fragmentation itself implies a TCP split).